.github/workflows/test-and-release.yml

```diff
@@ -0,0 +1,61 @@
+name: Test and Release
+on:
+  push:
+  pull_request:
+    types:
+      - opened
+      - reopened
+
+jobs:
+  test:
+    name: Test and Report
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: 16
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Test with Coverage
+        run: npm run test-with-coverage
+
+      - name: Upload to codecov
+        run: npm i -g codecov && codecov
+  release:
```

Comment:
There are ways of making workflow depending on each other, but this was the easiest solution.